shared/health-bindings/src/test/java/io/pravega/shared/health/bindings/HealthTests.java

```diff
@@ -0,0 +1,366 @@
+        return "http";
+    }
+
+    @After
+    public void tearDown() {
+        healthServiceManager.close();
+        client.close();
+        restServer.stopAsync();
+        restServer.awaitTerminated();
+    }
+
+
+    protected URI getURI(String path) {
+        return UriBuilder.fromPath(path)
+                .scheme(getURLScheme())
+                .host(serverConfig.getHost())
+                .port(serverConfig.getPort())
+                .build();
+    }
+
+    @Test
+    public void testHealth() throws Exception {
+        // Register the HealthIndicator.
+        healthServiceManager.getRoot().register(new StaticHealthyContributor());
+
+        URI streamResourceURI = UriBuilder.fromUri(getURI("/v1/health"))
+                .scheme(getURLScheme()).build();
+
+        URI serviceStatusURI = UriBuilder.fromUri(getURI("/v1/health/status"))
+                .scheme(getURLScheme()).build();
+        assertStatus(serviceStatusURI, HealthStatus.UP);
+        URI contributorStatusURI = UriBuilder.fromUri(getURI("/v1/health/status/" + StaticHealthyContributor.NAME))
+                .scheme(getURLScheme()).build();
+        // Assert the registered contributor has been initialized.
+        assertStatus(contributorStatusURI, HealthStatus.UP);
+
+        Response response = client.target(streamResourceURI).request().buildGet().invoke();
+        HealthResult healthResult = response.readEntity(HealthResult.class);
+        boolean found = false;
+        for (val child : healthResult.getChildren().entrySet()) {
+            found |= !child.getValue().getDetails().isEmpty();
+        }
+        Assert.assertTrue("HealthService should provide details.", found);
+    }
+
+    @Test
+    public void testContributorHealth() throws Exception {
+        // Register the HealthIndicator.
+        StaticHealthyContributor contributor = new StaticHealthyContributor();
+        healthServiceManager.getRoot().register(contributor);
+
+        // Wait for contributor initialization.
+        URI statusURI = UriBuilder.fromUri(getURI(String.format("/v1/health/status/%s", contributor.getName())))
+                .scheme(getURLScheme()).build();
+        assertStatus(statusURI, HealthStatus.UP);
+
+        URI streamResourceURI = UriBuilder.fromUri(getURI(String.format("/v1/health/%s", contributor.getName())))
+                .scheme(getURLScheme()).build();
+        Response response = client.target(streamResourceURI).request().buildGet().invoke();
+        HealthResult healthResult = response.readEntity(HealthResult.class);
+
+        Assert.assertEquals("Unexpected number of child contributors.",
+            0,
+            healthResult.getChildren().size());
+
+        Assert.assertEquals("Child HealthContributor failed to report its detail entry.",
+                1,
+                healthResult.getDetails().size());
+    }
+
+    // Be mindful that the results will be dependent on the StatusAggregatorRule used.
```

Comment:
I don't understand this comment. Is it a TODO? If not, then I don't think it belongs here - should it go next to some actual check you're making?

Reply:
Many of these comments are outdated and no longer valid, will make sure to check of references to outdated concepts.